chain/client/src/chunk_validation.rs

```diff
@@ -261,18 +267,76 @@ fn pre_validate_chunk_state_witness(
         )));
     }
 
+    let block = Chain::get_apply_chunk_block_context(
```

Comment:
This block context corresponds to the last block with non-missing chunk in this shard, so you should not use anything from that block to validate new transactions since it is too old. Main source of data for that should be `state_witness.chunk_header`